secboot/secboot_tpm.go

```diff
@@ -273,9 +276,86 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 	}
 }
 
+// FDERevealKeyRequest carries the operation and parameters for the
+// fde-reveal-key binary to support unsealing keys that were sealed
+// with the "fde-setup" hook.
+type FDERevealKeyRequest struct {
+	Op string `json:"op"`
+
+	SealedKey []byte `json:"sealed-key"`
+	KeyName   string `json:"key-name"`
+
+	// TODO: add VolumeName,SourceDevicePath later
+}
+
+// fdeRevealKeyRuntimeMax is the maximum runtime a fde-reveal-key can execute
+// XXX: what is a reasonable default here?
+var fdeRevealKeyRuntimeMax = "2m"
+
+// overridden in tests
+var fdeRevealKeyCommandExtra []string
+
+// fdeRevealKeyCommand returns a *exec.Cmd that is suitable to run
+// fde-reveal-key using systemd-run
+func fdeRevealKeyCommand() *exec.Cmd {
+	// TODO: put this into a new "systemd/run" package
+	cmd := exec.Command(
+		"systemd-run",
+		"--pipe", "--same-dir", "--wait", "--collect",
+		"--service-type=exec",
+		"--quiet",
+		// ensure we get some result from the hook within a
+		// reasonable timeout and output from systemd if
+		// things go wrong
+		fmt.Sprintf("--property=RuntimeMaxSec=%s", fdeRevealKeyRuntimeMax),
+		// this ensures we get useful output for e.g. segfaults
+		`--property=ExecStopPost=/bin/sh -c 'if [ "$EXIT_STATUS" != 0 ]; then echo "service result: $SERVICE_RESULT" 1>&2; fi'`,
```

Comment:
Quoting looks a bit unexpected here (but I guess I'm missing the trick), I would expect something like `--property=ExecStopPost="/bin/sh -c ...`; Anyway, the test below looks fine 👍